timon/probes.py

```diff
@@ -333,3 +362,79 @@ def probe_action(self):
 
 class DiskFreeProbe(Probe):
     pass
+
+
+class HttpJsonIntervalProbe(HttpProbe):
+    """
+    probe checking if a value is:
+        - between 2 values (example: "key1.key2:[0, 20]")
+        - greater than a value (example: "key1.key2.key3>60")
+        - lesser than a value (example: "key<20")
+        - equal to a value (example: "key1.key2:200")
+    """
+    script_module = "timon.scripts.http_json"
+
+    def __init__(self, **kwargs):
+        self.ok_rule = kwargs.pop('ok_rule', None)
+        self.warning_rule = kwargs.pop('warning_rule', None)
+        super().__init__(**kwargs)
+
+    def match_rule(self, rslt, rule):
+        rule_types = {
+            "equal_rule": re.compile(r"^(.*):([a-zA-Z0-9]+)$"),
+            "greater_rule": re.compile(r"^(.*)>(\d+)$"),
+            "lesser_rule": re.compile(r"^(.*)<(\d+)$"),
+            "interval_rule": re.compile(r"^(.*):\[(\d+),\s*(\d+)\]$"),
+            }
+
+        def define_rule(rule):
+            for rule_type, rule_rex in rule_types.items():
+                match = rule_rex.match(rule)
+                if match:
+                    return rule_type, match
+            return None
+
+        rule_type, match = define_rule(rule)
+        val = get_nested_field(match.groups()[0], rslt)
+        if rule_type == "equal_rule":
+            return str(val) == match.groups()[1]
```

Comment:
not sure but perhaps nicer to read and easier to change if yuou aver change the regexps.

> rule_val = match.groups()[1:]

And use then `rule_val[0]` / `rule_val[1]` in the if / elif chains